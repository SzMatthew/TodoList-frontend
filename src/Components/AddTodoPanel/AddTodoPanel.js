import React, {useEffect, useState} from 'react';
import {BsFillFlagFill} from 'react-icons/bs';
import {IconContext} from "react-icons";
import {useRef} from 'react';
import {Row, Col} from 'react-grid-system';
import './AddTodoPanel.scss';

const AddTodoPanel = ({setAddTaskOpen, addTaskOpen, AddTodo}) => {
    const [priority, setPriorty] = useState(0);
    const [newTodo, setNewTodo] = useState('');
    const todoRef = useRef();

    useEffect(() => todoRef.current.focus(), []);

    const AddNewTodo = () => {
        if (newTodo)
        {
            AddTodo(newTodo, priority);
            setNewTodo('');
        }
    }

    return(
        <Row className="add_todo_panel_wrapper">
            <Col xs={12}>
                <Row>
                    <Col lg={9} md={8} sm={12} xs={12}>
                        <input
                            type        = "text"
                            ref         = {todoRef}
                            placeholder = "e.g. Learn Portugese every 2 days"
                            value       = {newTodo}
                            onKeyPress  = {(event) => {if (event.charCode === 13 && newTodo) { AddNewTodo()}}}
                            onChange    = {(event) => setNewTodo(event.target.value)}
                        />
                    </Col>
                    <Col lg={3} md={4} sm={4} xs={6} className="priority-wrapper">
                        <div className={priority === 1 ? 'priority priorty-active' : 'priority'} onClick={() => setPriorty(1)}>
                            <IconContext.Provider value={{color: "#DE4C4A", size: "25px"}}> <BsFillFlagFill /></IconContext.Provider>
                        </div>
                        <div className={priority === 2 ? 'priority priorty-active' : 'priority'} onClick={() => setPriorty(2)}>
                            <IconContext.Provider value={{color: "#FB9814", size: "25px"}}><BsFillFlagFill /></IconContext.Provider>
                        </div>
                        <div className={priority === 3 ? 'priority priorty-active' : 'priority'} onClick={() => setPriorty(3)}>
                            <IconContext.Provider value={{color: "#4271B8", size: "25px"}}><BsFillFlagFill /></IconContext.Provider>
                        </div>
                        <div className={priority === 0 ? 'priority priorty-active' : 'priority'} onClick={() => setPriorty(4)}>
                            <IconContext.Provider value={{color: "#7F7F7F", size: "25px"}}> <BsFillFlagFill /></IconContext.Provider>
                        </div>
                    </Col>
                </Row>
            </Col>
            <Col xs={12}>
                <button type="button" disabled={!newTodo} className="red" onClick={AddNewTodo}>Add Task</button>
                <button type="button" onClick={() => setAddTaskOpen(!addTaskOpen)}>Cancel</button>
            </Col>
        </Row>
    )
}

export default AddTodoPanel;