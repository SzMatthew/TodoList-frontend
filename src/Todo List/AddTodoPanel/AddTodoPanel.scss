@import '../../Global.scss';

.add-new-input-wrapper{
    display: flex;

    input{
        width: 80%;
    }

    .priority-wrapper{
        margin: 5px 5px;
        display: flex;

        .priority{
            width: 30px;
            margin: 0px 2px;
            display: flex;
            justify-content: center;
            align-items: center;
            border-radius: 6px;
            border: 2px solid transparent;
            transition: 0.2s;

            &:hover{
                border: 2px solid $lightest_gey;

            }
        }

        .priorty-active{
            background-color: $mid_gey;

        }
    }
}