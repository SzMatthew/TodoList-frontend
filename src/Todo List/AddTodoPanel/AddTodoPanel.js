import React, {useEffect, useState} from 'react';
import {BsFillFlagFill} from 'react-icons/bs';
import {IconContext} from "react-icons";
import {useRef} from 'react';
import './AddTodoPanel.scss';

const AddTodoPanel = ({setAddTaskOpen, addTaskOpen, AddTodo}) => {
    const [priority, setPriorty] = useState(0);
    const [newTodo, setNewTodo] = useState('');
    const todoRef = useRef();

    useEffect(() => todoRef.current.focus(), []);

    const AddNewTodo = () => {
        if (newTodo)
        {
            AddTodo(newTodo, priority);
            setNewTodo('');
        }
    }

    return(
        <>
            <div className="add-new-input-wrapper">
                <input
                    type        = "text"
                    ref         = {todoRef}
                    placeholder = "e.g. Learn Portugese every 2 days"
                    value       = {newTodo}
                    onKeyPress  = {(event) => {if (event.charCode === 13 && newTodo) { AddNewTodo()}}}
                    onChange    = {(event) => setNewTodo(event.target.value)}
                />
                
                <div className="priority-wrapper">
                    <div className={priority === 1 ? 'priority priorty-active' : 'priority'} onClick={() => setPriorty(1)}>
                        <IconContext.Provider value={{color: "#DE4C4A", size: "25px"}}> <BsFillFlagFill /></IconContext.Provider>
                    </div>
                    <div className={priority === 2 ? 'priority priorty-active' : 'priority'} onClick={() => setPriorty(2)}>
                        <IconContext.Provider value={{color: "#FB9814", size: "25px"}}><BsFillFlagFill /></IconContext.Provider>
                    </div>
                    <div className={priority === 3 ? 'priority priorty-active' : 'priority'} onClick={() => setPriorty(3)}>
                        <IconContext.Provider value={{color: "#4271B8", size: "25px"}}><BsFillFlagFill /></IconContext.Provider>
                    </div>
                    <div className={priority === 0 ? 'priority priorty-active' : 'priority'} onClick={() => setPriorty(4)}>
                        <IconContext.Provider value={{color: "#7F7F7F", size: "25px"}}> <BsFillFlagFill /></IconContext.Provider>
                    </div>
                </div>
                
            </div>
            <button type="button" disabled={!newTodo} className="red" onClick={AddNewTodo}>Add Task</button>
            <button type="button" onClick={() => setAddTaskOpen(!addTaskOpen)}>Cancel</button>
        </>
    )
}

export default AddTodoPanel;