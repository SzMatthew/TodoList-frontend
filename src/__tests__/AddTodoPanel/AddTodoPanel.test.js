import {shallow, mount} from 'enzyme';
import AddTodoPanel from '../../Todo List/AddTodoPanel/AddTodoPanel';

describe('AddTodoPanel', () => {
    it('Renders', () => {
        const wrapper = shallow(<AddTodoPanel />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Set priority', () => {
        const wrapper = shallow(<AddTodoPanel />);
        wrapper.find('div.priority').first().simulate('click');
        setTimeout(() => {
            expect(wrapper.find('div.priority').first().hasClass('priority-active')).toEqual(true);
        }, 200);
        
    });

    it('Calls "Add TODO" function', () => {
        const AddTodo = jest.fn();
        const wrapper = mount(<AddTodoPanel AddTodo={AddTodo}/>);
        wrapper.find('button').first().simulate('click');
        expect(AddTodo).toBeCalled();
    });
});
