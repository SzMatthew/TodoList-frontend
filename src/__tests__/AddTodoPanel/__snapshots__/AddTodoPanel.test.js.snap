// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodoPanel Renders 1`] = `ShallowWrapper {}`;
